crates/edr_provider/src/requests/serde.rs

```diff
@@ -347,6 +347,47 @@ where
     Ok(result)
 }
 
+/// Value representing timestamp in seconds.
+///
+/// While technically the Yellow Paper allows specifying the timestamp as
+/// any scalar value, using `u64` in practice is more than enough.
+#[derive(Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord, Serialize)]
+pub struct Timestamp(pub u64);
+
+impl From<u64> for Timestamp {
+    fn from(value: u64) -> Self {
+        Self(value)
+    }
+}
+
+impl From<Timestamp> for u64 {
+    fn from(value: Timestamp) -> Self {
+        value.0
+    }
+}
+
+impl Timestamp {
+    pub fn as_u64(self) -> u64 {
+        self.0
+    }
+}
+
+impl<'de> Deserialize<'de> for Timestamp {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        // TODO: Accept only `0x`-prefixed hex strings (and possibly base 10 strings).
+        U64::deserialize(deserializer)
+            .map(|value| Timestamp(value.as_limbs()[0]))
+            .map_err(|_| {
+                serde::de::Error::custom(
+                    "Timestamp must be a positive integer or a string not exceeding 2^64 - 1",
```

Comment:
I'd say:

```suggestion
                    "Timestamp must be a non-negative number not exceeding 2^64 - 1",
```

- Non-negative, because `0` is accepted by `evm_increaseTime` (and it's probably fine to accept it).
- "Number" instead of "positive integer or a string" for conciseness. The fact that we accept both numbers and strings doesn't seem super relevant here.

Reply:
Done in https://github.com/NomicFoundation/edr/pull/494/commits/913edb7a511e996113c84e84560b13297029346a